src/bsmHandler.cpp

```diff
@@ -56,8 +56,15 @@ BSMHandler::BSMHandler(Quad::Ptr quad_ptr, const ConfigMap& conf ):
     json_{},
     vf_{ conf },
     idr_{ conf },
-    box_extension_{ 10.0 }
+    box_extension_{ 10.0 },
+    logger_{ logger }
 {
+    if (logger_ == nullptr) {
+        std::cout << "BSMHandler::BSMHandler(): Logger is null" << std::endl;
```

Comment:
If logger is null, should we return since we attempt to use it for trace statement directly below this and it'd blow up if the pointer was null?

Reply:
Good call! I've added a return statement here and modified the cout message.